gamemodes/terrortown/entities/entities/ttt_logic_role.lua

```diff
@@ -35,24 +37,36 @@ function ENT:KeyValue(key, value)
 end
 
 ---
--- @param string name
--- @param Entity|Player activator
--- @return[default=true] boolean
--- @realm shared
-function ENT:AcceptInput(name, activator)
+-- Called when another entity fires an event to this entity.
+-- @param string name The name of the input that was triggered
+-- @param Entity|Player activator The initial cause for the input getting triggered (e.g. the player who pushed a button)
+-- @param Entity caller The entity that directly triggered the input (e.g. the button that was pushed)
+-- @param string data The data passed
+-- @return[default=true] boolean Return true if the default action should be supressed
+-- @realm server
+function ENT:AcceptInput(name, activator, caller, data)
+	local cv_evil_roles = GetConVar("ttt2_rolecheck_all_evil_roles")
+
 	if name == "TestActivator" then
-		if IsValid(activator) and activator:IsPlayer() then
-			local activator_role = (GetRoundState() == ROUND_PREP) and ROLE_INNOCENT or activator:GetBaseRole()
+		if not IsValid(activator) or not activator:IsPlayer() then return end
+
+		local activator_role = (GetRoundState() == ROUND_PREP) and ROLE_INNOCENT
+			---
+			-- @realm server
+			or roles.GetByIndex(hook.Run("TTT2ModifyLogicCheckRole", ply, self, activator, caller, data) or ply:GetSubRole()):GetBaseRole()
```

Comment:
The end-bracket is in the wrong place.
Also why not move the ply:GetSubRole():GetBaseRole() into the hook. Thats the default case and therefore could be embedded directly inside or not?

Reply:
no, the hook can return any role that isn't a baserole. So we have to make sure we get the baserole here. And what do you mean with "The end-bracket is in the wrong place"?

Reply:
Its ply:GetSubRole() ):G.... not ply:GetSubRole():G.... )

Reply:
Yes, this is intended. What speaks against it?

Reply:
oh nvm..... I am confused by numbers and roles......

Reply:
Ply is not available here, I guess this needs to be activator there too.

Reply:
> Ply is not available here, I guess this needs to be activator there too.

yes, copy paste error